### ISSUE TITLE
Search Box Unfocusses During Virtual Keyboard Use

### ISSUE THREAD

**Issue by hepcat72** [popcorntime/popcorntime#225]
Operating System Version:

OS X 10.11

Popcorn Time Version:

Sorry, unknown at the time of typing this up

Download date:

Sorry, unknown at the time of typing this up
#### Expected Behaviour

I expected to be able to type text into the search box using the system's default virtual keyboard ("Keyboard Viewer").
#### Actual Behaviour

It seems that when the cursor gets a certain distance from the search box, the search box loses focus and virtual keyboard characters typed do not get entered into the search field.
#### Steps to reproduce the behavior
1. Put the popcorn time window at the top right of the screen
2. Go to System Preferences -> Keyboard and make sure "Show keyboard & character viewers in menu bar" is checked
3. Select "Show Keyboard Viewer" from the keyboard menu item at the top right of the screen
4. Drag the virtual keyboard to the bottom left of the screen
5. Click in the search box in Popcorn time
6. Type a search term using the virtual keyboard

**Comment by ChrisAlderson** [comment]
This is expected behaviour, the search bar will close when another window is selected. Is there any reason you need to use the virtual keyboard?

**Comment by hepcat72** [comment]
That's not what's happening.  Clicking the virtual keyboard (which is a special floating window) does not change window focus.  In fact, if I put the keyboard close-enough to the search field, I can use it fine, but if I put it in the lower left corner of the screen, the search field goes away as soon as the mouse reaches some proximity threshold (without even clicking or being over any window other than popcorn time).

To answer your question - my computer is a part of my entertainment system and I rarely ever break out the real keyboard because I don't want to leave the couch.  I have an MxAir mouse (think Wii Remote).  It's faster/easier, to just type a search in using the virtual keyboard than get up and use the real one.

**Comment by ChrisAlderson** [comment]
Alright you've made me understand the issue some more. I think this issue is Mac related, but sadly I don't have a Mac to test this on.

**Comment by hepcat72** [comment]
I rather doubt this is a Mac-specific issue or that you need a mac to test it. The behavior of the search field losing focus upon the cursor reaching a certain distance from it is one that is unique to the popcorn app. You don't need a virtual keyboard to reproduce the behavior. Just click in the search field and move the cursor to the lower left of the screen.

I've been using a virtual keyboard for probably over a decade and the popcorn app is the first app that has exhibited this behavior. I can click on any text field in any other app any move the cursor to all 4 corners of the screen and the focus will not change.

This being a Java app, I'm familiar with the code to change focus on hover. In fact, an app I'm developing had a focus change on hover issue in a search field that a fellow developer had intentionally coded. Apparently, when he clicks in a search field, he doesn't move the cursor away like I do, because he never had an issue entering text into it, but every time I used it, it was so annoying because immediately after clicking in the field, the focus would switch to another interface element. He was coding on windows and when I described what I was doing, he could reproduce it, no problem.

**Comment by ChrisAlderson** [comment]
I tried opening the search field and moved my cursor to the bottom left of my 1080p monitor. The search field was kept open.

I'm interested in which version you are running, you can check this by going to the 'About' screen by clicking on the info icon in the top right. On the About screen the version and git revision can be found in the bottom right corner. I'm asking this because the is not a `Java` app, this is written mainly in `JavaScript.

**Comment by hepcat72** [comment]
Well, given your question, I decided to install the latest version and the problem is gone. The version with the problem was 5.3.1.

Sorry to waste your time.

I installed the latest version on a different computer, BTW, and tested them each out just by clicking in the search box and hovering to the lower left corner (no virtual keyboard involved). The old version's search box loses focus and the new one doesn't.
